### PR TITLE
[#163615184] Update folder structure

### PR DESCRIPTION
**Pull request by Duncanian**
#### What does this PR do?
- Updates the existing folder structure

#### What are the relevant pivotal tracker stories?
[#163615184](https://www.pivotaltracker.com/story/show/163615184)

#### Screenshots (if appropriate)
screenshot 2019-01-31 at 21 55 49

### CODE REVIEW

File: app/controllers/Books.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+const db = require('../../database/models')
+const responses = require('../helpers/responses')
+
+class Books {
+  static async GetAllBooks(req, res) {
+    try {
+      const results = db.Book.findAll();
```
Comment (Rodgers-M) [inline app/controllers/Books.js]: you need to `await` this call to the database.

File: app/controllers/Books.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      const message = [200, 'Book retrieved successfully', true];
+      responses.response(res, message, results);
+    } catch (error) {
+      responses.handleError(error.toString(), 500, res);
+    }
+  }
+
+  static async CreateBook(req, res) {
+    try {
+      const results = db.Book.create({
```
Comment (Rodgers-M) [inline app/controllers/Books.js]: same case here, call to the db is async

File: app/app.js [hunk 3]
```diff
@@ -1,66 +1,19 @@
...
-    res.status(201).send(book)
-  })
-  .catch(err => res.status(400).send(err)))
-app.put('/api/books/:id', (req, res) => db.Book.findById(req.params.id)
-  .then((book) => {
-    book.addAuthors([1])
-    res.status(201).send(book)
-  })
-  .catch(err => res.status(400).send(err)))
+app.get('/api/v1/', (req, res) => res.status(200).send('welcome to books api'))
```
Comment (Rodgers-M) [inline app/app.js]: lines 13 to 17 can also be moved to the routes file, at the end of the file and the behavior will still be the same.

File: app/controllers/Books.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      const message = [201, 'Book created successfully', true];
+      responses.response(res, message, results);
+    } catch (error) {
+      responses.handleError(error.toString(), 500, res);
+    }
+  }
+
+  static async EditBook(req, res) {
+    try {
+      const results = await db.Book.findById(req.params.id);
```
Comment (Rodgers-M) [inline app/controllers/Books.js]: this implementation does not really do what the method name says, `editing a book`

File: app/helpers/responses.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+class Responses {
+  static response(res, message, result) {
```
Comment (Rodgers-M) [inline app/helpers/responses.js]: Good work here. however, you can do better in naming, calling `Responses.response` can be a bit confusing... how about naming the method something like `formatResponse` or `handleSuccess`... so calling `responses.formatResponse` makes more sense.

**Review by Rodgers-M — CHANGES_REQUESTED** [review]
Good job in the implementation, this looks great now.

A few things to note though
- any file that holds a class, its best practice to capitalize its name, same case with the class name. However, the method names in the class should be in `camelCase`, please check this in the `controllers` and the `helpers/responses` file.
- variable names should be descriptive enough. e.g instead of having `results` it is more intuitive to have `books` , `createdBook` and so on. same case with method names, `getOneBook` can be more explicit ...`getBookById` or `getBookByName`.... both of these methods return one book but take different parameters.

File: app/controllers/Books.js (hunk 1 above)
Comment (Duncanian) [reply app/controllers/Books.js]: Worked on it.

File: app/controllers/Books.js (hunk 2 above)
Comment (Duncanian) [reply app/controllers/Books.js]: Worked on it.

File: app/app.js (hunk 3 above)
Comment (Duncanian) [reply app/app.js]: Worked on it.

File: app/controllers/Books.js (hunk 4 above)
Comment (Duncanian) [reply app/controllers/Books.js]: Worked on it.

File: app/helpers/responses.js (hunk 5 above)
Comment (Duncanian) [reply app/helpers/responses.js]: Worked on it.

**Review by loicemeyo — APPROVED** [review]
This is great work @Duncanian. Wonderful comments and points of learning from @Rodgers-M

**Review by Rodgers-M — APPROVED** [review]

**Pull request merged**